packages/core/src/services/evaluations/prompt/index.ts

```diff
@@ -0,0 +1,107 @@
+import {
+  EvaluationDto,
+  EvaluationMetadataType,
+  EvaluationResultableType,
+  Workspace,
+} from '../../../browser'
+import { database } from '../../../client'
+import { LatitudeError, PromisedResult, Result } from '../../../lib'
+import { ProviderApiKeysRepository } from '../../../repositories'
+
+function valueInformation(evaluation: EvaluationDto) {
+  if (evaluation.resultType === EvaluationResultableType.Boolean) {
+    const base = 'A boolean value'
+    const valueDescriptions = [
+      evaluation.resultConfiguration.trueValueDescription
+        ? `true represents "${evaluation.resultConfiguration.trueValueDescription}"`
+        : undefined,
+      evaluation.resultConfiguration.falseValueDescription
+        ? `false represents "${evaluation.resultConfiguration.falseValueDescription}"`
+        : undefined,
+    ].filter(Boolean)
+
+    if (!valueDescriptions.length) return base
+    return `${base}, where ${valueDescriptions.join(' and ')}`
+  }
+
+  if (evaluation.resultType === EvaluationResultableType.Number) {
+    const base = `A number between ${evaluation.resultConfiguration.minValue} and ${evaluation.resultConfiguration.maxValue}`
+    const valueDescriptions = [
+      evaluation.resultConfiguration.minValueDescription
+        ? `${evaluation.resultConfiguration.minValue} represents "${evaluation.resultConfiguration.minValueDescription}"`
+        : undefined,
+      evaluation.resultConfiguration.maxValueDescription
+        ? `${evaluation.resultConfiguration.maxValue} represents "${evaluation.resultConfiguration.maxValueDescription}"`
+        : undefined,
+    ].filter(Boolean)
+
+    if (!valueDescriptions.length) return base
+    return `${base}, where ${valueDescriptions.join(' and ')}`
+  }
+
+  const base = 'A string value'
+  if (!evaluation.resultConfiguration.valueDescription) return base
+  return `${base} representing "${evaluation.resultConfiguration.valueDescription}"`
+}
+
+export async function getEvaluationPrompt(
+  {
+    workspace,
+    evaluation,
+  }: { workspace: Workspace; evaluation: EvaluationDto },
+  db = database,
+): PromisedResult<string, LatitudeError> {
+  if (evaluation.metadataType === EvaluationMetadataType.LlmAsJudgeAdvanced) {
+    return Result.ok(evaluation.metadata.prompt)
+  }
+
+  const providersRepo = new ProviderApiKeysRepository(workspace.id, db)
+  const providerResult = await providersRepo.find(
+    evaluation.metadata.providerApiKeyId,
+  )
+  if (providerResult.error) return providerResult
+
+  const resultSchema = {
+    [EvaluationResultableType.Boolean]: 'boolean',
+    [EvaluationResultableType.Number]: 'number',
+    [EvaluationResultableType.Text]: 'string',
+  } as const
+
+  return Result.ok(
+    `
+---
+provider: ${providerResult.value!.name}
+model: ${evaluation.metadata.model}
+temperature: 0.2
```

Comment:
? are we sure about this? I'd much rather evaluations be deterministic

Reply:
Hhmm yeah, I'll change it to `0`